lessons/02-first-ai-app/translations/README.fr.md

```diff
@@ -0,0 +1,501 @@
+}]; 
+
+// decide on the context window size 
+
+let max_tokens = 100; // Set the context window size 
+
+// Create the web request 
+
+const completion = await openai.chat.completions.create({
+    model: 'gpt-4',
+    messages: messages,
+    max_tokens: max_tokens
+}); 
+
+```
+
+> \[!ASTUCE]
+> Expérimentez avec différentes tailles de fenêtres de contexte pour voir comment cela affecte les réponses de l'IA.
+
+## Devoir - Construire un assistant en ingénierie
+
+Leonardo vous a soudainement demandé d'examiner de plus près le Time Beetle, il l’a regardé sous tous les angles, il l'a même secoué.
+
+**Leonardo :** J'ai besoin d'un assistant qui peut m'aider dans les calculs et la conception de la vis aérienne. Pouvez-vous me créer un assistant pour cela ?
+
+**Vous :** Bien sûr, je peux vous créer cela. Scarabée du Temps, on peut l'aider, non ?
+
+**Scarabée du Temps :** Oui, pas de problème, en fait la vis aérienne est l'une des inventions les plus fascinantes et visionnaires de Léonard. Conçue vers la fin des années 1480...
+
+**Vous :** Tout ce dont j'avais besoin, c'était un oui, gardez la leçon pour plus tard.
+
+**Scarabée du Temps :** Quelle impolitesse...
+
+**Vous :** Quoi ?
+
+**Scarabée du Temps :** Rien.
+
+![Aerial screw, Leonardo Da Vinci](https://raw.githubusercontent.com/microsoft/generative-ai-with-javascript/main/lessons/02-first-ai-app/assets/helicopter.jpg)
+
+> \[!NOTE]
+> La vis aérienne, également connue sous le nom de vis hélicoïdale, était destinée à décoller du sol en comprimant l'air. La conception de Léonard comportait un grand rotor en spirale en lin, rigidifié avec de l'amidon, et monté sur une plate-forme en bois. L'idée était qu'une équipe d'hommes ferait tourner des manivelles autour de la plate-forme, en la faisant tourner suffisamment vite pour obtenir une portance.
+>
+> Bien que Leonardo n’ait jamais réalisé une version grandeur nature de la vis aérienne, ses esquisses et notes fournissent des perspectives détaillées sur la manière dont il l'envisageait.
+>
+> Cependant, les scientifiques modernes s'accordent à dire que les matériaux disponibles à l'époque de Leonardo n'étaient ni assez solides ni assez légers pour rendre cela possible.
+>
+> Bien que son impraticabilité soit avérée, la vis aérienne reste un témoignage du génie de Léonard et de sa quête incessante d’innovation. Elle a jeté les bases des développements futurs dans l’aviation et continue d’inspirer les ingénieurs et inventeurs encore aujourd’hui.
+> [Lire la suite](https://en.wikipedia.org/wiki/Leonardo%27s_aerial_screw).
+
+Votre mission est de construire un assistant en ingénierie capable d’aider Leonardo dans les calculs et la conception de la vis aérienne.
+
+* Il doit être capable de générer du texte basé sur l’entrée utilisateur.
+
+* Vous devez définir un message système pour introduire l’assistant.
+
+Consultez l'[application d'exemple](/app/README.md) pour commencer.
+
+> \[!ASTUCE]
```

Comment:
This [!TIP] should not be translated. A good candidate for global rules

Reply:
all github alerts should be fixed now.